File: flow_test.py
#!/usr/bin/env python3
"""
Flow Tester for Mason Dashboard
Tests user journeys and navigation flows
"""

from playwright.sync_api import sync_playwright, expect
import json
from datetime import datetime
import time

BASE_URL = "http://localhost:3000"

def test_flows():
    results = {
        "agent_id": "FLOW-1",
        "completed_at": datetime.utcnow().isoformat() + "Z",
        "summary": {
            "flows_tested": 0,
            "flows_passed": 0,
            "flows_failed": 0,
            "issues_found": 0
        },
        "flows": []
    }

    with sync_playwright() as p:
        browser = p.chromium.launch(headless=True)
        context = browser.new_context(viewport={'width': 1920, 'height': 1080})
        page = context.new_page()

        # Enable console logging
        page.on("console", lambda msg: print(f"[CONSOLE {msg.type}] {msg.text}"))
        page.on("pageerror", lambda err: print(f"[PAGE ERROR] {err}"))

        # FLOW 1: Landing to Auth
        print("\n=== FLOW 1: Landing to Auth ===")
        flow1 = test_flow_1(page)
        results["flows"].append(flow1)
        results["summary"]["flows_tested"] += 1
        if flow1["status"] == "pass":
            results["summary"]["flows_passed"] += 1
        else:
            results["summary"]["flows_failed"] += 1
        results["summary"]["issues_found"] += len(flow1["issues"])

        # FLOW 2: Auth Page State
        print("\n=== FLOW 2: Auth Page State ===")
        flow2 = test_flow_2(page)
        results["flows"].append(flow2)
        results["summary"]["flows_tested"] += 1
        if flow2["status"] == "pass":
            results["summary"]["flows_passed"] += 1
        else:
            results["summary"]["flows_failed"] += 1
        results["summary"]["issues_found"] += len(flow2["issues"])

        # FLOW 3: Protected Route Redirect
        print("\n=== FLOW 3: Protected Route Redirect ===")
        flow3 = test_flow_3(page)
        results["flows"].append(flow3)
        results["summary"]["flows_tested"] += 1
        if flow3["status"] == "pass":
            results["summary"]["flows_passed"] += 1
        else:
            results["summary"]["flows_failed"] += 1
        results["summary"]["issues_found"] += len(flow3["issues"])

        # FLOW 4: Settings Navigation
        print("\n=== FLOW 4: Settings Navigation ===")
        flow4 = test_flow_4(page)
        results["flows"].append(flow4)
        results["summary"]["flows_tested"] += 1
        if flow4["status"] == "pass":
            results["summary"]["flows_passed"] += 1
        else:
            results["summary"]["flows_failed"] += 1
        results["summary"]["issues_found"] += len(flow4["issues"])

        browser.close()

    return results

def test_flow_1(page):
    """FLOW 1: Landing to Auth"""
    flow = {
        "name": "Landing to Auth",
        "status": "pass",
        "steps": [],
        "issues": []
    }

    try:
        # Step 1: Go to /
        print("Step 1: Navigating to /")
        page.goto(BASE_URL, wait_until="networkidle", timeout=10000)
        time.sleep(1)  # Let JS execute

        final_url = page.url
        print(f"Final URL: {final_url}")

        if "/auth/signin" in final_url:
            flow["steps"].append({
                "action": "goto /",
                "result": f"redirected to {final_url}",
                "status": "pass"
            })
            print("✓ Redirected to signin as expected")
        elif final_url == BASE_URL or final_url == BASE_URL + "/":
            # Check if landing content is shown
            page.screenshot(path="/tmp/flow1_landing.png", full_page=True)
            flow["steps"].append({
                "action": "goto /",
                "result": "shows landing page",
                "status": "pass"
            })
            print("✓ Landing page shown")
        else:
            flow["steps"].append({
                "action": "goto /",
                "result": f"unexpected redirect to {final_url}",
                "status": "fail"
            })
            flow["status"] = "fail"
            flow["issues"].append({
                "type": "unexpected_redirect",
                "severity": "medium",
                "description": f"Expected / or /auth/signin, got {final_url}",
                "step": "goto /",
                "screenshot": None
            })
            print(f"✗ Unexpected redirect to {final_url}")

    except Exception as e:
        flow["status"] = "fail"
        flow["steps"].append({
            "action": "goto /",
            "result": f"error: {str(e)}",
            "status": "fail"
        })
        flow["issues"].append({
            "type": "crash",
            "severity": "critical",
            "description": f"Failed to load root page: {str(e)}",
            "step": "goto /",
            "screenshot": None
        })
        print(f"✗ Error: {e}")

    return flow

def test_flow_2(page):
    """FLOW 2: Auth Page State"""
    flow = {
        "name": "Auth Page State",
        "status": "pass",
        "steps": [],
        "issues": []
    }

    try:
        # Step 1: Go to /auth/signin
        print("Step 1: Navigating to /auth/signin")
        page.goto(f"{BASE_URL}/auth/signin", wait_until="networkidle", timeout=10000)
        time.sleep(1)

        flow["steps"].append({
            "action": "goto /auth/signin",
            "result": "page loaded",
            "status": "pass"
        })
        print("✓ Auth page loaded")

        # Step 2: Check for GitHub sign-in button
        print("Step 2: Looking for GitHub sign-in button")
        page.screenshot(path="/tmp/flow2_auth.png", full_page=True)

        # Try different selectors for GitHub button
        github_button = None
        selectors = [
            "button:has-text('Sign in with GitHub')",
            "button:has-text('GitHub')",
            "a:has-text('Sign in with GitHub')",
            "a:has-text('GitHub')",
            "[data-testid='github-signin']"
        ]

        for selector in selectors:
            try:
                github_button = page.locator(selector).first
                if github_button.is_visible(timeout=1000):
                    print(f"✓ Found GitHub button with selector: {selector}")
                    break
            except:
                continue

        if github_button and github_button.is_visible():
            flow["steps"].append({
                "action": "check for GitHub button",
                "result": "GitHub sign-in button found",
                "status": "pass"
            })
            print("✓ GitHub sign-in button present")
        else:
            flow["steps"].append({
                "action": "check for GitHub button",
                "result": "GitHub sign-in button NOT found",
                "status": "fail"
            })
            flow["status"] = "fail"
            flow["issues"].append({
                "type": "missing_element",
                "severity": "critical",
                "description": "GitHub sign-in button not found on auth page",
                "step": "check for GitHub button",
                "screenshot": "/tmp/flow2_auth.png"
            })
            print("✗ GitHub sign-in button missing")

        # Step 3: Check for console errors
        print("Step 3: Checking for page errors")
        # Errors are logged via page.on listeners

    except Exception as e:
        flow["status"] = "fail"
        flow["steps"].append({
            "action": "test auth page",
            "result": f"error: {str(e)}",
            "status": "fail"
        })
        flow["issues"].append({
            "type": "crash",
            "severity": "critical",
            "description": f"Failed to test auth page: {str(e)}",
            "step": "test auth page",
            "screenshot": None
        })
        print(f"✗ Error: {e}")

    return flow

def test_flow_3(page):
    """FLOW 3: Protected Route Redirect"""
    flow = {
        "name": "Protected Route Redirect",
        "status": "pass",
        "steps": [],
        "issues": []
    }

    try:
        # Step 1: Go to /admin/backlog without auth
        print("Step 1: Navigating to /admin/backlog (protected)")
        page.goto(f"{BASE_URL}/admin/backlog", wait_until="networkidle", timeout=10000)
        time.sleep(1)

        final_url = page.url
        print(f"Final URL: {final_url}")

        # Step 2: Check if redirected or prompted
        if "/auth/signin" in final_url:
            flow["steps"].append({
                "action": "goto /admin/backlog without auth",
                "result": f"redirected to {final_url}",
                "status": "pass"
            })
            print("✓ Properly redirected to signin")
        elif "/admin/backlog" in final_url:
            # Check if auth prompt is shown
            page.screenshot(path="/tmp/flow3_protected.png", full_page=True)

            # Look for auth prompts or setup messages
            auth_indicators = [
                "Sign in",
                "Authentication required",
                "Please sign in",
                "GitHub",
                "Connect"
            ]

            page_text = page.content()
            has_auth_prompt = any(indicator.lower() in page_text.lower() for indicator in auth_indicators)

            if has_auth_prompt:
                flow["steps"].append({
                    "action": "goto /admin/backlog without auth",
                    "result": "page shows auth prompt",
                    "status": "pass"
                })
                print("✓ Auth prompt shown on protected route")
            else:
                flow["steps"].append({
                    "action": "goto /admin/backlog without auth",
                    "result": "no redirect and no auth prompt",
                    "status": "fail"
                })
                flow["status"] = "fail"
                flow["issues"].append({
                    "type": "broken_navigation",
                    "severity": "critical",
                    "description": "Protected route accessible without auth or prompt",
                    "step": "goto /admin/backlog without auth",
                    "screenshot": "/tmp/flow3_protected.png"
                })
                print("✗ No auth protection on protected route")
        else:
            flow["steps"].append({
                "action": "goto /admin/backlog without auth",
                "result": f"unexpected redirect to {final_url}",
                "status": "fail"
            })
            flow["status"] = "fail"
            flow["issues"].append({
                "type": "unexpected_redirect",
                "severity": "high",
                "description": f"Expected /auth/signin or auth prompt, got {final_url}",
                "step": "goto /admin/backlog without auth",
                "screenshot": None
            })
            print(f"✗ Unexpected redirect to {final_url}")

    except Exception as e:
        flow["status"] = "fail"
        flow["steps"].append({
            "action": "test protected route",
            "result": f"error: {str(e)}",
            "status": "fail"
        })
        flow["issues"].append({
            "type": "crash",
            "severity": "critical",
            "description": f"Failed to test protected route: {str(e)}",
            "step": "test protected route",
            "screenshot": None
        })
        print(f"✗ Error: {e}")

    return flow

def test_flow_4(page):
    """FLOW 4: Settings Navigation"""
    flow = {
        "name": "Settings Navigation",
        "status": "pass",
        "steps": [],
        "issues": []
    }

    settings_pages = [
        "/settings/database",
        "/settings/github",
        "/settings/api-keys"
    ]

    for settings_path in settings_pages:
        try:
            print(f"Testing navigation to {settings_path}")
            page.goto(f"{BASE_URL}{settings_path}", wait_until="networkidle", timeout=10000)
            time.sleep(1)

            final_url = page.url
            page.screenshot(path=f"/tmp/flow4_{settings_path.split('/')[-1]}.png", full_page=True)

            # Check if we stayed on the settings page or got redirected
            if settings_path in final_url:
                flow["steps"].append({
                    "action": f"goto {settings_path}",
                    "result": "page loaded successfully",
                    "status": "pass"
                })
                print(f"✓ {settings_path} loaded")
            elif "/auth/signin" in final_url:
                flow["steps"].append({
                    "action": f"goto {settings_path}",
                    "result": "redirected to signin (expected for protected route)",
                    "status": "pass"
                })
                print(f"✓ {settings_path} protected by auth")
            else:
                flow["steps"].append({
                    "action": f"goto {settings_path}",
                    "result": f"unexpected redirect to {final_url}",
                    "status": "fail"
                })
                flow["status"] = "fail"
                flow["issues"].append({
                    "type": "unexpected_redirect",
                    "severity": "medium",
                    "description": f"Expected {settings_path} or /auth/signin, got {final_url}",
                    "step": f"goto {settings_path}",
                    "screenshot": f"/tmp/flow4_{settings_path.split('/')[-1]}.png"
                })
                print(f"✗ Unexpected redirect from {settings_path} to {final_url}")

        except Exception as e:
            flow["status"] = "fail"
            flow["steps"].append({
                "action": f"goto {settings_path}",
                "result": f"error: {str(e)}",
                "status": "fail"
            })
            flow["issues"].append({
                "type": "crash",
                "severity": "high",
                "description": f"Failed to load {settings_path}: {str(e)}",
                "step": f"goto {settings_path}",
                "screenshot": None
            })
            print(f"✗ Error loading {settings_path}: {e}")

    return flow

if __name__ == "__main__":
    print("Starting Mason Dashboard Flow Tests")
    print(f"Base URL: {BASE_URL}\n")

    results = test_flows()

    # Print summary
    print("\n" + "="*60)
    print("FLOW TEST SUMMARY")
    print("="*60)
    print(f"Flows Tested: {results['summary']['flows_tested']}")
    print(f"Flows Passed: {results['summary']['flows_passed']}")
    print(f"Flows Failed: {results['summary']['flows_failed']}")
    print(f"Issues Found: {results['summary']['issues_found']}")
    print("="*60)

    # Save results
    import os
    os.makedirs("/home/jeffl/projects/mason/.claude/battle-test/results", exist_ok=True)

    output_file = "/home/jeffl/projects/mason/.claude/battle-test/results/FLOW-1.json"
    with open(output_file, 'w') as f:
        json.dump(results, f, indent=2)

    print(f"\nResults saved to: {output_file}")

    # Exit with error code if any flows failed
    exit(0 if results['summary']['flows_failed'] == 0 else 1)

File: scripts/flow_test.js
#!/usr/bin/env node

/**
 * Flow Tester for Mason Dashboard
 * Tests user journeys and navigation flows
 * Agent: FLOW-1
 */

import { chromium } from 'playwright';
import { writeFileSync, mkdirSync, existsSync } from 'node:fs';

const BASE_URL = process.env.BASE_URL || 'http://localhost:3000';

async function testFlows() {
  const results = {
    agent_id: 'FLOW-1',
    completed_at: new Date().toISOString(),
    summary: {
      flows_tested: 0,
      flows_passed: 0,
      flows_failed: 0,
      issues_found: 0,
    },
    flows: [],
  };

  const browser = await chromium.launch({ headless: true });
  const context = await browser.newContext({
    viewport: { width: 1920, height: 1080 },
  });
  const page = await context.newPage();

  // Console logging
  page.on('console', (msg) =>
    console.log(`[CONSOLE ${msg.type()}] ${msg.text()}`),
  );
  page.on('pageerror', (err) => console.error(`[PAGE ERROR] ${err}`));

  try {
    // FLOW 1: Landing to Auth
    console.log('\n=== FLOW 1: Landing to Auth ===');
    const flow1 = await testFlow1(page);
    results.flows.push(flow1);
    updateSummary(results, flow1);

    // FLOW 2: Auth Page State
    console.log('\n=== FLOW 2: Auth Page State ===');
    const flow2 = await testFlow2(page);
    results.flows.push(flow2);
    updateSummary(results, flow2);

    // FLOW 3: Protected Route Redirect
    console.log('\n=== FLOW 3: Protected Route Redirect ===');
    const flow3 = await testFlow3(page);
    results.flows.push(flow3);
    updateSummary(results, flow3);

    // FLOW 4: Settings Navigation
    console.log('\n=== FLOW 4: Settings Navigation ===');
    const flow4 = await testFlow4(page);
    results.flows.push(flow4);
    updateSummary(results, flow4);
  } finally {
    await browser.close();
  }

  return results;
}

function updateSummary(results, flow) {
  results.summary.flows_tested += 1;
  if (flow.status === 'pass') {
    results.summary.flows_passed += 1;
  } else {
    results.summary.flows_failed += 1;
  }
  results.summary.issues_found += flow.issues.length;
}

async function testFlow1(page) {
  const flow = {
    name: 'Landing to Auth',
    status: 'pass',
    steps: [],
    issues: [],
  };

  try {
    console.log('Step 1: Navigating to /');
    await page.goto(BASE_URL, { waitUntil: 'networkidle', timeout: 10000 });
    await page.waitForTimeout(1000);

    const finalUrl = page.url();
    console.log(`Final URL: ${finalUrl}`);

    if (finalUrl.includes('/auth/signin')) {
      flow.steps.push({
        action: 'goto /',
        result: `redirected to ${finalUrl}`,
        status: 'pass',
      });
      console.log('✓ Redirected to signin as expected');
    } else if (finalUrl === BASE_URL || finalUrl === BASE_URL + '/') {
      await page.screenshot({ path: '/tmp/flow1_landing.png', fullPage: true });
      flow.steps.push({
        action: 'goto /',
        result: 'shows landing page',
        status: 'pass',
      });
      console.log('✓ Landing page shown');
    } else {
      flow.steps.push({
        action: 'goto /',
        result: `unexpected redirect to ${finalUrl}`,
        status: 'fail',
      });
      flow.status = 'fail';
      flow.issues.push({
        type: 'unexpected_redirect',
        severity: 'medium',
        description: `Expected / or /auth/signin, got ${finalUrl}`,
        step: 'goto /',
        screenshot: null,
      });
      console.log(`✗ Unexpected redirect to ${finalUrl}`);
    }
  } catch (e) {
    flow.status = 'fail';
    flow.steps.push({
      action: 'goto /',
      result: `error: ${e.message}`,
      status: 'fail',
    });
    flow.issues.push({
      type: 'crash',
      severity: 'critical',
      description: `Failed to load root page: ${e.message}`,
      step: 'goto /',
      screenshot: null,
    });
    console.log(`✗ Error: ${e.message}`);
  }

  return flow;
}

async function testFlow2(page) {
  const flow = {
    name: 'Auth Page State',
    status: 'pass',
    steps: [],
    issues: [],
  };

  try {
    console.log('Step 1: Navigating to /auth/signin');
    await page.goto(`${BASE_URL}/auth/signin`, {
      waitUntil: 'networkidle',
      timeout: 10000,
    });
    await page.waitForTimeout(1000);

    flow.steps.push({
      action: 'goto /auth/signin',
      result: 'page loaded',
      status: 'pass',
    });
    console.log('✓ Auth page loaded');

    console.log('Step 2: Looking for GitHub sign-in button');
    await page.screenshot({ path: '/tmp/flow2_auth.png', fullPage: true });

    // Try different selectors for GitHub button
    const selectors = [
      'button:has-text("Sign in with GitHub")',
      'button:has-text("GitHub")',
      'a:has-text("Sign in with GitHub")',
      'a:has-text("GitHub")',
      '[data-testid="github-signin"]',
      'button:has-text("Continue with GitHub")',
    ];

    let found = false;
    for (const selector of selectors) {
      try {
        const element = page.locator(selector).first();
        if (await element.isVisible({ timeout: 1000 })) {
          console.log(`✓ Found GitHub button with selector: ${selector}`);
          found = true;
          break;
        }
      } catch (e) {
        // Continue to next selector
      }
    }

    if (found) {
      flow.steps.push({
        action: 'check for GitHub button',
        result: 'GitHub sign-in button found',
        status: 'pass',
      });
      console.log('✓ GitHub sign-in button present');
    } else {
      flow.steps.push({
        action: 'check for GitHub button',
        result: 'GitHub sign-in button NOT found',
        status: 'fail',
      });
      flow.status = 'fail';
      flow.issues.push({
        type: 'missing_element',
        severity: 'critical',
        description: 'GitHub sign-in button not found on auth page',
        step: 'check for GitHub button',
        screenshot: '/tmp/flow2_auth.png',
      });
      console.log('✗ GitHub sign-in button missing');
    }
  } catch (e) {
    flow.status = 'fail';
    flow.steps.push({
      action: 'test auth page',
      result: `error: ${e.message}`,
      status: 'fail',
    });
    flow.issues.push({
      type: 'crash',
      severity: 'critical',
      description: `Failed to test auth page: ${e.message}`,
      step: 'test auth page',
      screenshot: null,
    });
    console.log(`✗ Error: ${e.message}`);
  }

  return flow;
}

async function testFlow3(page) {
  const flow = {
    name: 'Protected Route Redirect',
    status: 'pass',
    steps: [],
    issues: [],
  };

  try {
    console.log('Step 1: Navigating to /admin/backlog (protected)');
    await page.goto(`${BASE_URL}/admin/backlog`, {
      waitUntil: 'networkidle',
      timeout: 10000,
    });
    await page.waitForTimeout(1000);

    const finalUrl = page.url();
    console.log(`Final URL: ${finalUrl}`);

    if (finalUrl.includes('/auth/signin')) {
      flow.steps.push({
        action: 'goto /admin/backlog without auth',
        result: `redirected to ${finalUrl}`,
        status: 'pass',
      });
      console.log('✓ Properly redirected to signin');
    } else if (finalUrl.includes('/admin/backlog')) {
      await page.screenshot({
        path: '/tmp/flow3_protected.png',
        fullPage: true,
      });

      // Look for auth indicators
      const content = await page.content();
      const authIndicators = [
        'Sign in',
        'Authentication required',
        'Please sign in',
        'GitHub',
        'Connect',
      ];

      const hasAuthPrompt = authIndicators.some((indicator) =>
        content.toLowerCase().includes(indicator.toLowerCase()),
      );

      if (hasAuthPrompt) {
        flow.steps.push({
          action: 'goto /admin/backlog without auth',
          result: 'page shows auth prompt',
          status: 'pass',
        });
        console.log('✓ Auth prompt shown on protected route');
      } else {
        flow.steps.push({
          action: 'goto /admin/backlog without auth',
          result: 'no redirect and no auth prompt',
          status: 'fail',
        });
        flow.status = 'fail';
        flow.issues.push({
          type: 'broken_navigation',
          severity: 'critical',
          description: 'Protected route accessible without auth or prompt',
          step: 'goto /admin/backlog without auth',
          screenshot: '/tmp/flow3_protected.png',
        });
        console.log('✗ No auth protection on protected route');
      }
    } else {
      flow.steps.push({
        action: 'goto /admin/backlog without auth',
        result: `unexpected redirect to ${finalUrl}`,
        status: 'fail',
      });
      flow.status = 'fail';
      flow.issues.push({
        type: 'unexpected_redirect',
        severity: 'high',
        description: `Expected /auth/signin or auth prompt, got ${finalUrl}`,
        step: 'goto /admin/backlog without auth',
        screenshot: null,
      });
      console.log(`✗ Unexpected redirect to ${finalUrl}`);
    }
  } catch (e) {
    flow.status = 'fail';
    flow.steps.push({
      action: 'test protected route',
      result: `error: ${e.message}`,
      status: 'fail',
    });
    flow.issues.push({
      type: 'crash',
      severity: 'critical',
      description: `Failed to test protected route: ${e.message}`,
      step: 'test protected route',
      screenshot: null,
    });
    console.log(`✗ Error: ${e.message}`);
  }

  return flow;
}

async function testFlow4(page) {
  const flow = {
    name: 'Settings Navigation',
    status: 'pass',
    steps: [],
    issues: [],
  };

  const settingsPages = [
    '/settings/database',
    '/settings/github',
    '/settings/api-keys',
  ];

  for (const settingsPath of settingsPages) {
    try {
      console.log(`Testing navigation to ${settingsPath}`);
      await page.goto(`${BASE_URL}${settingsPath}`, {
        waitUntil: 'networkidle',
        timeout: 10000,
      });
      await page.waitForTimeout(1000);

      const finalUrl = page.url();
      const screenshotName = settingsPath.split('/').pop();
      await page.screenshot({
        path: `/tmp/flow4_${screenshotName}.png`,
        fullPage: true,
      });

      if (finalUrl.includes(settingsPath)) {
        flow.steps.push({
          action: `goto ${settingsPath}`,
          result: 'page loaded successfully',
          status: 'pass',
        });
        console.log(`✓ ${settingsPath} loaded`);
      } else if (finalUrl.includes('/auth/signin')) {
        flow.steps.push({
          action: `goto ${settingsPath}`,
          result: 'redirected to signin (expected for protected route)',
          status: 'pass',
        });
        console.log(`✓ ${settingsPath} protected by auth`);
      } else {
        flow.steps.push({
          action: `goto ${settingsPath}`,
          result: `unexpected redirect to ${finalUrl}`,
          status: 'fail',
        });
        flow.status = 'fail';
        flow.issues.push({
          type: 'unexpected_redirect',
          severity: 'medium',
          description: `Expected ${settingsPath} or /auth/signin, got ${finalUrl}`,
          step: `goto ${settingsPath}`,
          screenshot: `/tmp/flow4_${screenshotName}.png`,
        });
        console.log(
          `✗ Unexpected redirect from ${settingsPath} to ${finalUrl}`,
        );
      }
    } catch (e) {
      flow.status = 'fail';
      flow.steps.push({
        action: `goto ${settingsPath}`,
        result: `error: ${e.message}`,
        status: 'fail',
      });
      flow.issues.push({
        type: 'crash',
        severity: 'high',
        description: `Failed to load ${settingsPath}: ${e.message}`,
        step: `goto ${settingsPath}`,
        screenshot: null,
      });
      console.log(`✗ Error loading ${settingsPath}: ${e.message}`);
    }
  }

  return flow;
}

// Main execution
console.log('Starting Mason Dashboard Flow Tests');
console.log(`Base URL: ${BASE_URL}\n`);

testFlows()
  .then((results) => {
    console.log('\n' + '='.repeat(60));
    console.log('FLOW TEST SUMMARY');
    console.log('='.repeat(60));
    console.log(`Flows Tested: ${results.summary.flows_tested}`);
    console.log(`Flows Passed: ${results.summary.flows_passed}`);
    console.log(`Flows Failed: ${results.summary.flows_failed}`);
    console.log(`Issues Found: ${results.summary.issues_found}`);
    console.log('='.repeat(60));

    // Save results
    const outputDir = '.claude/battle-test/results';
    if (!existsSync(outputDir)) {
      mkdirSync(outputDir, { recursive: true });
    }

    const outputFile = `${outputDir}/FLOW-1.json`;
    writeFileSync(outputFile, JSON.stringify(results, null, 2));

    console.log(`\nResults saved to: ${outputFile}`);

    // Exit with error code if any flows failed
    process.exit(results.summary.flows_failed === 0 ? 0 : 1);
  })
  .catch((e) => {
    console.error('Fatal error:', e.message);
    process.exit(1);
  });
